tools/redcap2linst.php

```diff
@@ -36,17 +38,37 @@
 
 $headers     = fgetcsv($fp);
 $instruments = [];
+$badMap = 0;
+$mapped = 0;
 while ($row = fgetcsv($fp)) {
     $inst = $row[1];
     if (!isset($instruments[$inst])) {
         $instruments[$inst] = [];
     }
-    $linstFormat = toLINST($row[3], $row[0], $row[4], $row[5]);
+    $fieldname = $row[0];
+    if ($trimName) {
+        $formname = $row[1];
+        if (strpos($fieldname, $formname) !== 0) {
+            $badMap++;
+            fwrite(STDERR, "Field $fieldname does not have form name $formname as a prefix\n");
+        } else {
+            // debug
+            //$oldname = $fieldname;
+            $mapped++;
+            $fieldname = preg_replace("/^" . preg_quote($formname) . "(_*)/", "", $fieldname);
```

Comment:
@driusan does this just remove all underscores?

Reply:
It removes the form name from the start of the field name and removes the underscores that are left at the front after trimming (ie if it's the bmi and the field is bmi_weight_kg it trims `bmi_` instead of `bmi` so that the fieldname that's left isn't `_weight_kg`